core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -798,13 +829,9 @@ class Partition(val topicPartition: TopicPartition,
      *
      **/
     val candidateReplicas = inSyncReplicas - leaderReplica
-
-    val laggingReplicas = candidateReplicas.filter(r =>
-      r.logEndOffset != leaderReplica.logEndOffset && (time.milliseconds - r.lastCaughtUpTimeMs) > maxLagMs)
-    if (laggingReplicas.nonEmpty)
-      debug("Lagging replicas are %s".format(laggingReplicas.map(_.brokerId).mkString(",")))
-
-    laggingReplicas
+    val currentTimeMs = time.milliseconds()
```

Comment:
Yeah, I guess we're inconsistent about this. Does `time.milliseconds()` have a side effect? I guess not because time moves whether or not you invoke `milliseconds()`. But another way to view it is that time doesn't move until you call `milliseconds()`, so the movement of time is the side effect. Viewing it that way accounts for `autoTickMs` in `MockTime`.